torch/csrc/autograd/profiler.h

```diff
@@ -97,13 +98,8 @@ enum class EventKind : uint16_t {
 };
 
 struct TORCH_API Event final {
-  Event(EventKind kind, std::string name, uint16_t thread_id, bool record_cuda)
-  : owned_name_(new std::string(std::move(name)))
-  , name_ptr_(owned_name_->c_str())
-  , kind_(kind)
-  , thread_id_(thread_id) { record(record_cuda); }
-  Event(EventKind kind, const char* name, uint16_t thread_id, bool record_cuda)
-  : name_ptr_(name)
+  Event(EventKind kind, const StringView& name, uint16_t thread_id, bool record_cuda)
```

Comment:
nit: it is possible to avoid the refcount:
StringView name,
... name_(std::move(name))